jmh-generator-annprocess/src/main/java/org/openjdk/jmh/generators/annotations/APGeneratorDestinaton.java

```diff
@@ -54,7 +55,16 @@ public InputStream getResource(String resourcePath) throws IOException {
 
     @Override
     public Writer newClass(String className) throws IOException {
-        return processingEnv.getFiler().createSourceFile(className).openWriter();
+        return newClass(className, null);
+    }
+
+    @Override
+    public Writer newClass(String className, String originatingClassNames) throws IOException {
+        if(originatingClassNames != null) {
+            return processingEnv.getFiler().createSourceFile(className).openWriter();
+        } else {
+            return processingEnv.getFiler().createSourceFile(className, originatingElements).openWriter();
```

Comment:
I don't think this even compiles ;)

Reply:
That's my fault for trying to make a "quick fix" while traveling for the holidays. Sorry about that, correct change is pushed. I'm running through my manual verifications in the meanwhile just to make sure nothing funky changed.